### DISCUSSION TITLE
Why is this GPL?

### DISCUSSION THREAD

**Discussion by 8bitprodigy** [CuarzoSoftware/Louvre#11] [General]
This is a library. Why not use a more permissive license, such as MIT, BSD, or even LGPL?

**Comment by ehopperdietzel** [comment]
I understand that there are individuals/companies who want to keep their software private, whether it's for selling purposes or other reasons. However, at the same time, it raises concerns for me, especially in this field where it's quite possible that they might be extracting information from users. While this might sound a bit paranoid, it's not something to be taken lightly. Therefore, it's a dilemma that I need to reflect on. In the meantime, it will remain under GPL3, which doesn't preclude you from generating income by using it.

**Reply by probonopd** [reply]
Well, I am working on a BSD licensed open source desktop environment. Hence I cannot use GPL3 software (even tough my project is  and I am not making money from it).

**Reply by ehopperdietzel** [reply]
Which license would you recommend that restricts proprietary use while allowing you to use it? I'm asking sincerely, as I'm not familiar with all the specific details and conditions of all licenses. If you can point me to one, it would be very helpful.

**Reply by probonopd** [reply]
Well, I guess that's logically not possible, because I chose the BSD license that does allow proprietary use while you are looking to prevent proprietary use. All I am saying is that there are noncommercial open source projects out there (like mine) that don't want to restrict commercial, proprietary use. Like FreeBSD, for example.

**Reply by ehopperdietzel** [reply]
Well, then we didn't get anywhere, hahaha. The solution seems simple: either I change my license or you change yours. Can I ask why you use that one? Don't you mind its proprietary use?

**Reply by probonopd** [reply]
Well, I am using a BSD license because my desktop environment is for FreeBSD and I don't want to restrict what one can or cannot do with my software.

**Reply by 8bitprodigy** [reply]
I mean, if you don't want people using it, GPL is fine, I guess. lmao

More permissive licenses are always going to gain more traction, and individuals, especially those who use Linux, aren't stupid. They can make decisions for themselves, like whether or not they want to incorporate proprietary software that could potentially extract information on them into their system. You don't need to act like some sort of savior.

**Reply by ehopperdietzel** [reply]
Yes, now that I think about it, what you're saying makes complete sense. I'll likely go with MPL; as I understand it, it allows use in proprietary software, but any modifications to the project files themselves must be shared with the community. Therefore, it seems like a good choice.

**Reply by 8bitprodigy** [reply]
MPL isn't bad if that's what you're going for. Far more terse than the LGPL.

**Reply by ehopperdietzel** [reply]
I finally decided to go with MIT, but I first need to get rid of FreeImage, which uses GPL hahaha

**Reply by 8bitprodigy** [reply]
Gotcha.

**Reply by 8bitprodigy** [reply]
I'm mostly familiar with gamedev libraries, so my recommendation would be something like sdl or raylib for image loading. Maybe libpng if you're cool with supporting just png images.

**Reply by ehopperdietzel** [reply]
Thanks :), I'll look into them. But I think I'll just use libpng and libjpeg since they seem to be in every distro. Later, if the developer needs other formats, they could include libraries on their own.

**Reply by 8bitprodigy** [reply]
Those are good choices. Very popular,  and very permissively licensed.

**Comment by dblsaiko** [comment]
> Which license would you recommend that restricts proprietary use while allowing you to use it? I'm asking sincerely, as I'm not familiar with all the specific details and conditions of all licenses. If you can point me to one, it would be very helpful.

LGPL, maybe. It allows linking any software against your library but modifications to your library must stay licensed as LGPL or GPL. https://choosealicense.com/licenses/lgpl-3.0/

It's the same license Qt and also KDE Frameworks is licensed under, for example.

**Reply by ehopperdietzel** [reply]
Thank you, very useful website. I've been giving this more thought, and I believe it's ultimately up to the users to decide whether they want to use proprietary software or not. Therefore, I will look for a less restrictive license that allows proprietary projects to be distributed. However, if any improvements or changes are made to the library itself, I want to require that they be shared with the community. So I think the MPL license would be a good choice.
